### PR TITLE
add new function in service for updating replicas

### PR DESCRIPTION
**Pull request by akshay-duplo**
## **Type**
enhancement, documentation

___

## **Description**
- Introduced a new method `update_replicas` in `service.py` to allow updating the number of replicas for a service.
- Added a new CLI argument `REPLICAS` to specify the number of replicas when using the command line.
- Updated the `CONTRIBUTING.md` with detailed instructions on version bumping, dependency installation, package building, and documentation.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>service.py</strong><dd><code>Add Method to Update Service Replicas</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/duplo_resource/service.py

<li>Added a new method <code>update_replicas</code> to update the number of replicas <br>for a service.<br> <li> The method takes service name and replica count as arguments.<br> <li> Utilizes the existing <code>duplo.post</code> method to send the update request.<br> <li> Handles service not found error by raising <code>DuploError</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/duploctl/pull/43/files#diff-de4bdab4d945f3f4c737be6a44e3c3c11850f302766a911d4bc30229766398ab">+21/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>args.py</strong><dd><code>Introduce REPLICAS Argument for CLI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
src/duplocloud/args.py

<li>Introduced a new command-line argument <code>REPLICAS</code> to specify the number <br>of replicas for a service.<br> <li> The argument is of type integer and is identified by <code>-r</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/duploctl/pull/43/files#diff-10408bee68a44ad34e4226ec91f20e3277d2d637fd8e17d3482fea817838ec96">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CONTRIBUTING.md</strong><dd><code>Update Contributing Guidelines</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
CONTRIBUTING.md

<li>Updated instructions for version bumping, including commands for <br>patch, major, and minor updates.<br> <li> Added instructions for installing dependencies using pip for both bash <br>and zsh shells.<br> <li> Included a new section on building the package and generating <br>artifacts.<br> <li> Updated the section on Semver with Setuptools SCM command, providing <br>commands to get the current version.<br> <li> Emphasized the importance of adding wiki documentation for new <br>subcommands.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/duploctl/pull/43/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055">+22/-9</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by kferrone — APPROVED** [review]

File: src/duplo_resource/service.py [hunk 1]
```diff
@@ -34,6 +34,27 @@ def find(self,
...
+    try:
+      tenant_id = self.tenant["TenantId"]
+      data = {
+        "Replicas": replica,
+        "Name": name
+      }
+      response = self.duplo.post(f"subscriptions/{tenant_id}/ReplicationControllerChangeAll", data)
+      if response.json() is None:
+        return {"message": f"Successfully updated replicas for service '{name}'"} 
+    except IndexError:
```
Comment (kferrone) [inline src/duplo_resource/service.py:55]: How does this happen? I wouldn't expect to see this at this level, something below should catch these and then rethrowing as DuploError

File: src/tests/test_commander.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import pytest 
-import unittest
+# import pytest
```
Comment (kferrone) [inline src/tests/test_commander.py:1]: pretty sure these are required for pytest to pick them up. What happened?

Comment (akshay-duplo) [reply src/tests/test_commander.py:1]: These were unused by the code

File: src/duplo_resource/service.py (hunk 1 above)
Comment (akshay-duplo) [reply src/duplo_resource/service.py:55]: Api gives null response on successful post call. So i added a condition

**Pull request merged**